internal/ignition/templates.go

```diff
@@ -82,3 +84,8 @@ func toJsonFunc(data interface{}) (result string, err error) {
 	result = string(text)
 	return
 }
+
+// toBase64 is a template function that encodes the given data using Base64.
+func toBase64Func(data string) string {
+	return base64.StdEncoding.EncodeToString([]byte(data))
```

Comment:
I'm not a fan of `makeExecuteTemplateFunc` calling `buffer.String()` because it assumes the file we wish to embed is a utf-8 encoded text file, which limits us to only store textual files. Ideally makeExecuteTemplateFunc would return a byte array, and toBase64Func would receive a byte array, and so the []byte(data) conversion on this line would not be needed

Reply:
No concrete need for binary files at the moment, just striving to do what's right without unnecessary assuming encoding where it's not needed.